src/document/docUtil.ts

```diff
@@ -0,0 +1,53 @@
+import { SelectionItem } from '../store';
+
+/**
+ * Finds the closest ancestor DOM element with the specified class.
+ */
+export function findClosestElWithClass(element: Element | null, className: string): Element | null {
+    for (let el = element; el && el !== ((document as unknown) as Element); el = el.parentNode as Element | null) {
+        if (el.classList && el.classList.contains(className)) {
+            return el;
+        }
+    }
+
+    return null;
+}
+
+/**
+ * Returns the page element and page number that the element is on.
+ * If not found return null/-1
+ */
+export function getPageInfo(element: Element | null): { pageEl: Element | null; page: number } {
+    const pageEl = findClosestElWithClass(element, 'page') || null;
+    const page = parseInt((pageEl && pageEl.getAttribute('data-page-number')) || '-1', 10);
+
+    return { pageEl, page };
+}
+
+export function getRange(): Range | null {
+    const selection = window.getSelection();
+    if (!selection || selection.isCollapsed) {
+        return null;
+    }
+
+    return selection.getRangeAt(0);
```

Comment:
Will we always only care about the first range of the selection?

Reply:
Yeah, I don't think pdf.js lets user select two ranges at the same time.

Reply:
Oh no, I just found firefox can select multiple ranges. I'll ask product/design how to solve this.

Reply:
Product confirms we don't support multiple ranges in one selection.